### ISSUE TITLE
For 6.X release, build universal bundle for macOS to support Apple Silicon

### ISSUE THREAD

**Issue by PaulWessel** [GenericMappingTools/gmt#4442]
**Description of the desired feature**

With Apple transitioning to its own CPUs (RiSC), there will be a several year transition period where our macOS bundle applications need to be _universal_, i.e., contain binaries for both RiSC and Intel.  This is similar to back in the mid 2000 when Apple transition from Motorola/IBM to Intel and our Makefiles at the time could build these "fat" executables.  The first M1-supported macBooks will soon be in the wild.

**Comment by PaulWessel** [comment]
I think we are close to being able to do this.  Since I wrote this, the ffmpeg (libxvid) issue seems to have been fixed (I am being specific to MacPorts here - it may have been fixed earlier for other distributions).  The new version of geos (3.9.1) builds on Apple Silicon so I assume GDAL now can be properly built - once the ARC CI that Macports uses passes the test I think it will be merged into their master distro shortly.

**Comment by PaulWessel** [comment]
The geos library has now been fixed to 3.9.1 so GDAL builds and GMT can now be built with all the usual dependencies.  Running the tests gave a handful of failures not seen on my Intel macs:

```
	157 - doc/scripts/GMT_tut_11.sh (Failed)
	164 - doc/scripts/GMT_tut_18.sh (Failed)
	233 - doc/examples/ex49/ex49.sh (Failed)
	237 - doc/examples/anim02/anim02.sh (Failed)
	316 - test/gmt_core/testsuite.sh (Failed)
	427 - test/grdfill/constfill.sh (Failed)
	428 - test/grdfill/nnfill.sh (Failed)
	430 - test/grdfill/splinefill.sh (Failed)
	445 - test/grdimage/autointense.sh (Failed)
	452 - test/grdimage/grdcyclic.sh (Failed)
	521 - test/grdview/autointense.sh (Failed)
	536 - test/greenspline/gspline_4.sh (Failed)
	677 - test/pscoast/oblsuite_N.sh (Failed)
```

I will have to see what is going on.  Many of these are not subtle, such as the two tutorial figs missing surfaces.  Some are related to grdview.  More later.

**Comment by PaulWessel** [comment]
Something funny happened when downloading the 05m topo and one side had all zeros.  Removing that file and rerunning the scripts helped.  Now I have

```
	233 - doc/examples/ex49/ex49.sh (Failed)
	237 - doc/examples/anim02/anim02.sh (Failed)
	316 - test/gmt_core/testsuite.sh (Failed)
	452 - test/grdimage/grdcyclic.sh (Failed)
	677 - test/pscoast/oblsuite_N.sh (Failed)
```

and these are more subtle, along the lines we may see between Windows and macOS output.  Will look a bit further on these.

**Comment by PaulWessel** [comment]
I did not have DCW 2.0.0 installed so it downloaded it to ~/.gmt/server/geography/dcw and that caused testsuite.sh to fail.  Installing it traditionally eliminated this error, so down to four failures:

```
	233 - doc/examples/ex49/ex49.sh (Failed)
	452 - test/grdimage/grdcyclic.sh (Failed)
	677 - test/pscoast/oblsuite_N.sh (Failed)
```

**ex49** fails because the median age/depth symbols are just different, and not just by a whisker.   **grdcyclic** again as roundoff I suspect.  **oblsuite_N.sh** also has ghostlines related to gridlines, i.e., roundoff.  I suspect this has to do with subtle differences in how floor, ceil, round etc works for Intel vs ARM.  I see no large errors here.

**Comment by weiji14** [comment]
Not sure if this is the best place to post this, but just as an update on the conda-forge migration status, there are still several upstream dependencies that we're waiting on before we can get a **GMT arm-osx build** on https://anaconda.org/conda-forge/gmt/files. Keeping track at https://conda-forge.org/status/#armosxaddition.

![image](https://user-images.githubusercontent.com/23487320/118048581-950a9400-b3d0-11eb-8ae4-cbe5816d01ae.png)

- [x] https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/1434
- [x] https://github.com/conda-forge/dcw-gmt-feedstock/pull/5
- [x] https://github.com/conda-forge/gshhg-gmt-feedstock/pull/7
- [ ] https://github.com/conda-forge/qt-feedstock/pull/185
- [ ] https://github.com/conda-forge/ghostscript-feedstock/pull/22
- [ ] gnuplot
- [ ] graphicsmagick
- [ ] gmt :tada:

**Comment by PaulWessel** [comment]
I think the effort involved to build a universal bundle far outweighs the ease of building separate Intel and ARM bundles, so I am closing this issue.
